src/DotNetWorker.Core/Context/RetryContext.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Exposes information about retry acvitity for the event that triggered
+    /// the current function invocation.
+    /// </summary>
+    public abstract class RetryContext
+    {
+        /// <summary>
+        /// The the retry count for the current event.
+        /// </summary>
+        public abstract int RetryCount { get; }
+
+        /// <summary>
+        /// The maximum number of retry attempts that will be made by the trigger/host
+        /// before the event is considered undeliverable.
+        /// </summary>
+        public abstract int MaxRetryCount { get; }
+
```

Comment:
Should we surface the retry exception here too?

Reply:
Thought about this, need to look at what exception details are actually passed in (if it's truly the original exception, with an appropriate stack trace and something actionable), otherwise, we might be better off simply surfacing the error message. 

I had this flagged for a future PR, but would be open to adding now if we feel it's critical to have that.

Reply:
Makes sense!

Reply:
I don't think we need to block on this. Let's flag this for a follow up.